File: Roboti/app/src/main/java/cz/muni/fi/umimecesky/adapterlistener/LevelAdapter.java
package cz.muni.fi.umimecesky.adapterlistener;

import android.app.Activity;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.akexorcist.roundcornerprogressbar.RoundCornerProgressBar;

import java.util.List;

import cz.muni.fi.umimecesky.R;
import cz.muni.fi.umimecesky.pojo.RaceConcept;

public class LevelAdapter extends BaseAdapter {

    private List<RaceConcept> list;
    private Activity activity;

    public LevelAdapter(Activity activity, List<RaceConcept> list) {
        super();
        this.activity = activity;
        this.list = list;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        final ViewHolder holder;

        if (convertView == null) {
            convertView = LayoutInflater.from(activity).inflate(R.layout.column_category_race, parent, false);

            holder = new ViewHolder();
            holder.sectionName = (TextView) convertView.findViewById(R.id.sectionName);
            holder.currentLevel = (TextView) convertView.findViewById(R.id.currentLevel);
            holder.progressBar = (RoundCornerProgressBar) convertView.findViewById(R.id.levelProgress);
            holder.progressBar.setMax(list.get(position).getNumberOfLevels() - 1);
            Log.i("max bar", String.valueOf(holder.progressBar.getMax()));
            convertView.setTag(holder);
        } else {
            holder = (ViewHolder) convertView.getTag();
        }

        RaceConcept raceConcept = list.get(position);
        Log.d("race concept", raceConcept + "");
        holder.sectionName.setText(raceConcept.getName());
        setProgressBar(holder, raceConcept);
        return convertView;
    }

    private void setProgressBar(ViewHolder holder, RaceConcept raceConcept) {
        RoundCornerProgressBar progressBar = holder.progressBar;
        TextView textCurrentLevel = holder.currentLevel;
        int currentLevel = raceConcept.getCurrentLevel();
        textCurrentLevel.setText(String.valueOf(currentLevel));
        currentLevel--; //progress bar starts at 0, level at 1
        progressBar.setProgress(currentLevel);
        progressBar.setSecondaryProgress(currentLevel + 1);
    }

    private class ViewHolder {
        TextView sectionName;
        TextView currentLevel;
        RoundCornerProgressBar progressBar;
    }

    @Override
    public int getCount() {
        return list.size();
    }

    @Override
    public Object getItem(int position) {
        return list.get(position);
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }
}

File: Roboti/app/src/main/java/cz/muni/fi/umimecesky/adapterlistener/LevelAdapter.kt
package cz.muni.fi.umimecesky.adapterlistener

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.akexorcist.roundcornerprogressbar.RoundCornerProgressBar
import cz.muni.fi.umimecesky.R
import cz.muni.fi.umimecesky.pojo.RaceConcept

class LevelAdapter(val items: List<RaceConcept>, val listener: (RaceConcept) -> Unit) :
        RecyclerView.Adapter<LevelAdapter.ViewHolder>() {

    fun ViewGroup.inflate(layoutRes: Int): View =
            LayoutInflater.from(context).inflate(layoutRes, this, false)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = ViewHolder(parent.inflate(R.layout.column_category_race))

    override fun onBindViewHolder(holder: ViewHolder, position: Int) = holder.bind(items[position], listener)

    override fun getItemCount(): Int = items.size

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        private val sectionName = itemView.findViewById<TextView>(R.id.sectionName)
        private val currentLevel = itemView.findViewById<TextView>(R.id.currentLevel)
        private val progressBar = itemView.findViewById<RoundCornerProgressBar>(R.id.levelProgress)

        fun bind(item: RaceConcept, listener: (RaceConcept) -> Unit) = with(itemView) {
            sectionName.text = item.name
            val currentLevelNumber = item.getCurrentLevel()
            currentLevel.text = currentLevelNumber.toString()
            progressBar.max = (item.numberOfLevels - 1).toFloat()
            progressBar.progress = (currentLevelNumber - 1).toFloat()
            progressBar.secondaryProgress = currentLevelNumber.toFloat()
            setOnClickListener { listener(item) }
        }
    }

}

